rfc/egress-billing.md

```diff
@@ -0,0 +1,309 @@
+this should be done in background jobs and the read logic should read from whatever data store these tasks are using to manage
+rate limits.
+
+```mermaid
+sequenceDiagram
+  Gateway-)Accounting Service: I received a request for bafk...7fi
+  Accounting Service->>Content Claims Service: Give me claims relevant to this request
+  Content Claims Service-->>Accounting Service: Here are the claims I can find
+  Accounting Service->>Rate Limiting Service: Increment rate limit usage for this claim
+  Rate Limiting Service-->>Accounting Service: OK, done
+```
+
+#### Bulk Configuration
+
+NOTE: The following requires support for "delegations without inline proofs" as specified in UCAN 1.0
+
+If users would like to set rate limits on a set of CIDs in a single operation, they can create an intermediary "principle"
+to which they can address delegations - in this example we will call this principle `did:key:zGroup`. 
+
+We first create a content commitment addressed to the group:
+
+```js
+{ // bafy..group
+  "iss": "did:key:zAlice",
+  "aud": "did:key:zGroup",
+  "exp": 1716235987 // restrict to a month
+  "cmd": "/assert/location",
+  // Subject is storage node because they own that resource
+  "sub": "did:web:asia.web3.storage",
+  "pol": [
+    // Request origin header must be example.com
+    ["==", ".headers['origin']", "example.com"],
+    ["==", ".query.token", "zrptvx"]
+  ],  
+  "meta": { "proof": { "/": "bafy...site" } }
+}
+```
+
+and then create an open-ended delegation from the group to the gateway, with the same `pol` (TODO: or more restrictive?) fields:
+
+```js
+{ // bafy..access
+  "iss": "did:key:zGroup",
+  "aud": "did:web:w3s.link",
+  "exp": 1716235987 // restrict to a month
+  "cmd": "/assert/location",
+  // Subject is storage node because they own that resource
+  "sub": "did:web:asia.web3.storage",
+  "pol": [
+    // Request origin header must be example.com
+    ["==", ".headers['origin']", "example.com"],
+    ["==", ".query.token", "zrptvx"]
+  ]
+}
+```
+
+For any given CID, the gateway will be able to look up the CID-specific content commitment as well as the
+group-to-gateway content commitment and combine them to create a chain rooted in the original content commitment
+provided by the w3up service on content upload. The gateway should take care to check that each link in the 
+chain is valid (though this will hopefully be handled by the UCAN tooling).
+
+```mermaid
+sequenceDiagram
+  Alice->>Blob Service: Please store bafk...7fi
+  Blob Service-->>Alice: Stored - you can present bafy...site to fetch bafk...7fi
+  Alice-)Group: You can present bafy...group to fetch bafk...7fi
+  Group-)Gateway: You can present bafy..access to fetch bafk...7fi
+```
+
+Note that `bafy..access` alone will not be enough to authorize retrieval
+of the content - the accounting system will need to combine it with
+a CID-specific content commitment that has been delegated to `Group`.
```

Comment:
I'm not following how this works, I think you need to show the "CID-specific content commitment that has been delegated to `Group`"